lib/opFns.js

```diff
@@ -310,8 +310,8 @@ module.exports = {
     memStore(runState, offset, word, new BN(0), new BN(32))
   },
   MSTORE8: function (offset, byte, runState) {
-    // NOTE: we're using a 'trick' here to get the least significant byte
-    byte = byte.toArrayLike(Buffer, 'le', 1)
+    byte = byte.toArrayLike(Buffer, 'be', 32)
+    byte = byte.slice(byte.length - 1)
```

Comment:
Can you extract a case from the tests where this caused an issue? I'd like to figure out what is the actual reason.

Reply:
``node tests/tester.js -s --test='randomStatetest237' --jsontrace``

In Node console:

```node
const utils = require('ethereumjs-util')
const BN = utils.BN
var bn = new BN('fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe', 'hex')
bn.toArrayLike(Buffer, 'le', 1)
```

gives

```
Error: byte array longer than desired length
    at assert (/Users/hdrewes/Documents/DEV/EthereumJS/ethereumjs-vm/node_modules/bn.js/lib/bn.js:6:21)
    at BN.toArrayLike (/Users/hdrewes/Documents/DEV/EthereumJS/ethereumjs-vm/node_modules/bn.js/lib/bn.js:527:5)
    at repl:1:4
    at ContextifyScript.Script.runInThisContext (vm.js:44:33)
    at REPLServer.defaultEval (repl.js:239:29)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:440:10)
    at emitOne (events.js:120:20)
    at REPLServer.emit (events.js:210:7)
```

Reply:
Actually this is failing for every Buffer with length > 1, e.g. also for ``var bn = new BN('fffe', 'hex')``.

Reply:
Ah true, never end the end added the truncation support to bn.js.

Reply:
Could still do `bn.andl(0xff).toArrayLike(Buffer, 'le', 1)`